includes/blocks/class-sensei-block-helpers.php

```diff
@@ -16,44 +16,51 @@ class Sensei_Block_Helpers {
 
 
 	/**
-	 * Build CSS classes (for named colors) and inline styles from block attributes.
+	 * Build CSS classes and inline styles from block attributes.
 	 *
-	 * @param array $block  Block.
-	 * @param array $colors Color attributes and their style property.
+	 * @param array $block_attributes  The block attributes array with the format$attribute_key => $attribute_value.
 	 *
-	 * @return array Colors CSS classes and inline styles.
+	 * @return array {
+	 *     An array of classes and styles.
+	 *
+	 *     @type string $css_classes   An array of classes.
+	 *     @type string $inline_styles An array of styles.
+	 * }
 	 */
-	public static function build_color_styles( $block, $colors = [] ) {
-		$block_attributes = $block['attributes'] ?? [];
-		$attributes       = [
-			'css_classes'   => [],
-			'inline_styles' => [],
-		];
-
-		$colors = array_merge(
-			[
-				'textColor'       => 'color',
-				'backgroundColor' => 'background-color',
-			],
-			$colors
-		);
+	public static function build_block_styles( $block_attributes ) {
+		$inline_styles = [];
+		$css_classes   = [];
 
-		foreach ( $colors as $color => $style ) {
-
-			$named_color  = $block_attributes[ $color ] ?? null;
-			$custom_color = $block_attributes[ 'custom' . ucfirst( $color ) ] ?? null;
-
-			if ( $custom_color || $named_color ) {
-				$attributes['css_classes'][] = sprintf( 'has-%s', $style );
-			}
-			if ( $named_color ) {
-				$attributes['css_classes'][] = sprintf( 'has-%s-%s', $named_color, $style );
-			} elseif ( $custom_color ) {
-				$attributes['inline_styles'][] = sprintf( '%s: %s;', $style, $custom_color );
+		foreach ( $block_attributes as $attribute => $value ) {
+			switch ( $attribute ) {
```

Comment:
I meant here to only move that existing bit of code for the fontSize inside the style builder method — keep the `foreach` for the colors, and just add the fontSize support after that. 

That color builder was more generic and configurable, because there are cases (for module block styles now) where we want to have for example a `mainColor` that is mapped to either background-color or border-color, depending on the style.

Reply:
Yep. The thing is that the previous implementation was specific to colors and was a bit hard to follow, this is the reason that I didn't use it in the first place. I think that having specific branches for each different attribute makes the code more readable.

Reply:
Ready to approve apart from this bit, can we keep the original color mapping logic here? It's the same logic as the Gutenberg editor does to produce these colors, adding the custom prefix, so I'd rather have something here that handles that. It's also needed for some upcoming PRs.

Reply:
Reverted here e873a4f40f6d1eec05b2833ceb21cb638a9ea90c